airbyte-integrations/connectors/source-lever/source_lever/source.py

```diff
@@ -0,0 +1,267 @@
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import BasicHttpAuthenticator
+
+def _auth_from_config(config):
+    try:
+        if config["api_key"]:
+            return BasicHttpAuthenticator(username=config["api_key"], password=None, auth_method="Basic")
+        else:
+            print("Auth type was not configured properly")
+            return None
+    except Exception as e:
+        print(f"{e.__class__} occurred, there's an issue with credentials in your config")
+        raise e
+    
+
+class SourceLever(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        try:
+            authenticator = _auth_from_config(config)
+            _ = authenticator.get_auth_header()
+        except Exception as e:
+            return False, str(e)
+        return True, None
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        authenticator = _auth_from_config(config)
+        # TODO: use extract_start_date
+        return [
+            Opportunities(authenticator=authenticator),
+            Offers(authenticator=authenticator),
+            Feedback(authenticator=authenticator),
+            Interviews(authenticator=authenticator),
+            Applications(authenticator=authenticator),
+            Requisitions(authenticator=authenticator),
+            Users(authenticator=authenticator),
+            Stages(authenticator=authenticator),
+            Postings(authenticator=authenticator),
+            ArchiveReasons(authenticator=authenticator),
+            Panels(authenticator=authenticator),
+            Tags(authenticator=authenticator),
+            Sources(authenticator=authenticator),
+            RequisitionFields(authenticator=authenticator),
+            Notes(authenticator=authenticator)
+        ]
+
+# Basic full refresh stream
+class LeverStream(HttpStream, ABC):
+    page_size = 100
+    stream_params = {}
+    
+    API_VERSION = "v1"
+    base_url = "https://api.lever.co"
+
+    @property
+    def url_base(self) -> str:
+        return f"{self.base_url}/{self.API_VERSION}/"
+    
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        response_data = response.json()
+        if response_data.get("hasNext"):
+            return {"offset": response_data["next"]}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"limit": self.page_size}
+        params.update(self.stream_params)
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+
+    def parse_response(self, response: requests.Response, stream_slice:Mapping[str, Any],  **kwargs) -> Iterable[Mapping]:
+        yield from response.json()["data"]
+
+class Opportunities(LeverStream):
```

Comment:
Should we add a link to the Opportunities API docs?

https://hire.lever.co/developer/documentation#list-all-opportunities

Or 

https://hire.lever.co/developer/documentation#retrieve-a-single-opportunity

Reply:
Yeap added